### PR TITLE
script: build new zephyr module from a new STM32Cube

### PR DESCRIPTION
**Pull request by FRASTM**
For stm32 series, get the version tree found on
https://github.com/STMicroelectronics/STM32Cube
and apply patches to update module to the latest version

run with `$ python ./scripts/update_stm32_package.py stm32xx`

Signed-off-by: Francois Ramu <francois.ramu@st.com>

### CODE REVIEW

**Comment by FRASTM** [comment]
fixing above remarks in scripts
Rename generated update directory with a more explicit name, like 'zephyr_module', but removed
STM32Cube repo is kept for future use
Also mention full path in script log : "Cloning into 'STM32CubeF2'... comes from git command itself
Note that some lines remains > 80

**Comment by FRASTM** [comment]
include the patch on _hal_conf.h coming

**Comment by FRASTM** [comment]
run pep8

**Comment by FRASTM** [comment]
apply patch from the previous version on top of the new version

**Comment by FRASTM** [comment]
rebase

**Comment by FRASTM** [comment]
In case of git error  when applying the patch, a log file is generated and the files are let uncommited

**Comment by FRASTM** [comment]
This version builds the patch for the zephyr module ith git diff --ignore-space-at-eol 
Then the new cube cube version is copied and this patch is applied to get the new updated dzephyr module.
use the -f option to force the commit : _python zephyr_module.py stm32f3 -f_

**Comment by FRASTM** [comment]
Adding one script to build the _CMakeList.txt_ only
usage:  `python make_cmakelist.py stm32f3`

File: script/zephyr_module.py [hunk 1]
```diff
@@ -0,0 +1,313 @@
...
+        print 'unknown option'
+if args['a2']:
+    if 'c' in args['a2']:
+        option_clean = True
+    elif 'stm32' in args['a2']:
+        module_serie = args['a2']
+    elif 'f' in args['a2']:
+        option_force = True
+    elif 'f' in args['a2']:
+        option_force = True
```
Comment (erwango) [inline script/zephyr_module.py]: Not sure about this line (copy paste from previous line)

Comment (FRASTM) [reply script/zephyr_module.py]: fixed

**Comment by FRASTM** [comment]
This version will preserve the module_stm32xx.log file in case of merge error
Files are already added to git but not merge in case of error (except if '-f' option is set).
Use '-c' option to keep the working 'zephyr_module' repo.

**Comment by FRASTM** [comment]
Change the title of the commit message to _stm32cube: update stm32XX to version V1.11.0_
Keep the .rej files (not removed) un-committed
Add a warning to check the content of the patch list in the README file

**Comment by carlescufi** [comment]
Oh I forgot. Please use `series` and not `serie` in the code and commit messages.

**Comment by FRASTM** [comment]
rename _script_ dir to _scripts_ dir 
rename to **update_stm32_package.py**
write _series_ in the commit

**Comment by erwango** [comment]
@carlescufi can you review again?

**Pull request merged**